openmc/bounding_box.py

```diff
@@ -99,19 +106,64 @@ def extend(self, padding_distance: float) -> BoundingBox:
 
         Returns
         -------
-        An enlarged bounding box
+        An expanded bounding box
+        """
+        self[0] -= padding_distance
+        self[1] += padding_distance
+        return self
+
+    def expand(self, other_box: BoundingBox) -> BoundingBox:
+        """Expand the box to contain another box
+
+        Parameters
+        ----------
+        other_box : BoundingBox
+            The box used to resize this box
+
+        Returns
+        -------
+        An expanded bounding box
+        """
+        self[0][:] = np.minimum(self.lower_left, other_box.lower_left)
+        self[1][:] = np.maximum(self.upper_right, other_box.upper_right)
```

Comment:
With the fix in `__setitem__`, I think the following suffices:
```suggestion
        self[0] = np.minimum(self.lower_left, other_box.lower_left)
        self[1] = np.maximum(self.upper_right, other_box.upper_right)
```